feat(casperf/lvol): add lvol uri type

tiagolobocastro opened the pull request:
This allows us to run casperf against an lvol, example: sudo casperf
"lvol:///lvol-1?size=220GiB&lvs=lvs:///pool-1?mode=purge&disk=aio:///dev/sda" -b 128ki -q 1

tiagolobocastro committed:
feat(casperf/lvol): add lvol uri type

This allows us to run casperf against an lvol, example:
sudo casperf
"lvol:///lvol-1?size=220GiB&lvs=lvs:///pool-1?mode=purge&disk=aio:///dev/sda" -b 128ki -q 1

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

hrudaya21 commented on io-engine/src/bdev/lvs.rs:
```diff
@@ -0,0 +1,279 @@
+//! operation but for testing and benchmarking.
+//!
+//! # Uri
+//! lvol:///$name?size=$size&lvs=$lvs
```
How this uri is used in io-engine-client CLI ? 
`./io-engine-client pool create lvol:///$name?size=$size&lvs=$lvs`

hrudaya21 replied:
Sorry.. understood.. It is part of PR description :)

tiagolobocastro committed:
fix(lvs/error): handle missing errors during import/create

Some error codes not being properly mapped or handled making it impossible to create
or import a pool in certain situations.
This ensures we handle errors like already exists properly which allows the control-plane
to ensure pool creation/import.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge

tiagolobocastro committed:
fix(pcie): spdk requires nvme path id

When destroying a pcie nvme bdev we must not specify the nvme path id.
Also it was noticed that when destroy fails we end up with no alias which
could cause some panics. For this particular case we now add the alias
back if the bdev is still present though we need a more robust fix..

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
refactor(lvs): add error reason to pool import error

Existing import error was confusing with name being replaced with a message
at times!
Add a reason which can help make it clear.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
bors cancel

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
bors merge
